### PR TITLE
fix(core,docs): add local rtl service for notification component

### LINKED ISSUE #5515: Notification component: RTL is not working properly
Description: RTL is not working 

Expected: RTL should work properly 

Attached the screenshot.

Screenshot 2021-05-28 at 14 15 34

### PR DESCRIPTION
**Pull request by N1XUS**
#### Please provide a link to the associated issue.
Closes #5515 

#### Please provide a brief summary of this pull request.
Added locally-scoped RTL service to the notification service and generated component.
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- n/a update `README.md`
- n/a [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- n/a Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/notification/notification/notification.component.ts [hunk 1]
```diff
@@ -90,14 +96,23 @@ export class NotificationComponent extends AbstractFdNgxClass implements OnInit,
...
         this._setNotificationConfig(notificationConfig);
     }
 
     /** @hidden */
     ngOnInit(): void {
+
+        this._subscriptions.add(
+            this._rtlService?.rtl.subscribe(isRtl => {
+                this._dir = isRtl ? 'rtl' : 'ltr';
+                this.cdRef.markForCheck();
```
Comment (Smiranin) [inline libs/core/src/lib/notification/notification/notification.component.ts:112]: Do we have some code style guide how to use subscription? 
I mean, the subscription and the success handler must be on the same line, or vice versa, the subscription must always start on a new line, and each handler in the subscription?

**Review by Smiranin — APPROVED** [review]

File: libs/core/src/lib/notification/notification/notification.component.ts (hunk 1 above)
Comment (N1XUS) [reply libs/core/src/lib/notification/notification/notification.component.ts:112]: I don't think that we have such code style. It was copy-pasted from different component, and as I see, majority of components use the same code style.

File: libs/core/src/lib/notification/notification-service/notification.service.ts [hunk 2]
```diff
@@ -13,7 +13,10 @@ export class NotificationService {
 
     public containerRef: ComponentRef<NotificationContainer>;
 
-    constructor(private dynamicComponentService: DynamicComponentService) {}
+    constructor(
```
Comment (InnaAtanasova) [inline libs/core/src/lib/notification/notification-service/notification.service.ts:16]: Should be hidden

File: libs/core/src/lib/notification/notification-service/notification.service.ts [hunk 3]
```diff
@@ -13,7 +13,10 @@ export class NotificationService {
...
+        private dynamicComponentService: DynamicComponentService,
```
Comment (InnaAtanasova) [inline libs/core/src/lib/notification/notification-service/notification.service.ts:17]: should be prefixed with `_` as it's private

File: libs/core/src/lib/notification/notification/notification.component.ts [hunk 4]
```diff
@@ -90,14 +96,23 @@ export class NotificationComponent extends AbstractFdNgxClass implements OnInit,
         private cdRef: ChangeDetectorRef,
         private _router: Router,
         @Optional() private notificationConfig: NotificationConfig,
-        @Optional() private notificationRef: NotificationRef
+        @Optional() private notificationRef: NotificationRef,
```
Comment (InnaAtanasova) [inline libs/core/src/lib/notification/notification/notification.component.ts:99]: should be prefixed with `_`

File: libs/core/src/lib/notification/notification/notification.component.ts (hunk 1 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/notification/notification/notification.component.ts:112]: cdRef is also private and should be prefixed with `_`

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

File: libs/core/src/lib/notification/notification-service/notification.service.ts [hunk 5]
```diff
@@ -43,11 +49,11 @@ export class NotificationService {
         let notificationComponentRef: ComponentRef<NotificationComponent>;
 
         // Create Notification Component
-        notificationComponentRef = this.dynamicComponentService.createDynamicComponent(
+        notificationComponentRef = this._dynamicComponentService.createDynamicComponent(
             content,
             NotificationComponent,
             notificationConfig,
-            { services: [notificationService, notificationConfig] }
+            { services: [notificationService, notificationConfig, this._rtlService] }
```
Comment (dimamarksman) [inline libs/core/src/lib/notification/notification-service/notification.service.ts:56]: I suggest using the `injector` option instead.
`services` option is so unclear (not obvious what will be used as an injection token in this case). I would remove `services` option at all as the `injector` option is much more flexible and transparent

Comment (N1XUS) [reply libs/core/src/lib/notification/notification-service/notification.service.ts]: Done. Please check

**Review by dimamarksman — APPROVED** [review]

**Pull request merged**